api/views/recommendation.py

```diff
@@ -0,0 +1,113 @@
+import random
+
+from django.db import models
+from django.db.models import Count, Q
+from rest_framework import mixins, viewsets
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.decorators import action
+from rest_framework.response import Response
+
+from api.models import Song, SongEntry, WishlistSong
+from api.serializers.recommendation import RecommendationSerializer
+from api.views.authentication.token import TokenBasedAuthentication
+
+
+class RecommendationViewSet(
+    mixins.RetrieveModelMixin,
+    viewsets.GenericViewSet,
+):
+    """
+    API endpoint that constructs and serves a list of 6 songs: Wishlist > Likes > random recommendations.
+    """
+
+    serializer_class = RecommendationSerializer
+    queryset = WishlistSong.objects.all()
+    authentication_classes = [
+        SessionAuthentication,
+        TokenBasedAuthentication,
+    ]
```

Comment:
You can remove this entire `authentication_class` block, that was only needed for the wishlist endpoint so that the ios app can connect to it. If you remove the whole block, it will revert to the default which we've specified to be `SessionAuthentication`

Reply:
Okay, thanks!